plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/manager/VmwareStorageManagerImpl.java

```diff
@@ -1464,4 +1440,45 @@ private String deleteDir(String dir) {
     private static String getVolumeRelativeDirInSecStroage(long volumeId) {
         return "volumes/" + volumeId;
     }
+
+    /**
+     * This method is to check if a given TaskInfo Object is valid( and has name and entity name assigned).It return true if TaskInfo Object is valid and false otherwise.
+     *
+     * @param TaskInfo
+     *        info
+     * @return boolean(true or false)
+     **/
+    private boolean isvalidTaskInfoObj(TaskInfo info){
+        return !(info == null || info.getEntityName() == null || info.getName() == null);
+    }
+
+    private boolean waitForRunningTaskOnVM(String vmName, String taskName, VmwareContext context) throws Exception {
+        try {
+            ManagedObjectReference taskmgr = context.getServiceContent().getTaskManager();
+            List<ManagedObjectReference> tasks = context.getVimClient().getDynamicProperty(taskmgr, "recentTask");
+
+            for (ManagedObjectReference taskMor : tasks) {
+                TaskInfo info = (TaskInfo)(context.getVimClient().getDynamicProperty(taskMor, "info"));
+
+                if (!isvalidTaskInfoObj(info)) {
```

Comment:
Do you mean remove the “isvalidTaskInfoObj” and add that condition to the (if)?
I believe the way it is coded now, it is ok. It enables the writing of test cases and proper java documentation.